check/src/main/java/ru/tinkoff/qa/neptune/check/EvaluateAndMatchAction.java

```diff
@@ -0,0 +1,29 @@
+package ru.tinkoff.qa.neptune.check;
+
+import org.hamcrest.Matcher;
+
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.nonNull;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static ru.tinkoff.qa.neptune.core.api.steps.StoryWriter.toGet;
+
+class EvaluateAndMatchAction<T, R> extends MatchAction<T, R> {
+
+    private final Function<T, R> eval;
+
+    EvaluateAndMatchAction(Matcher<? super R> criteria, String description, Function<T, R> eval) {
```

Comment:
It does not need at all, but in order to keep parameter position, swap description and criteria.